lib/load-simple-icons.js

```diff
@@ -5,18 +5,23 @@ const { svg2base64 } = require('./svg-helpers')
 
 function loadSimpleIcons() {
   const simpleIcons = {}
+  // As of v5 the exported keys are the svg slugs
+  // Historically, Shields has supported logo specification via
+  // name, name with spaces replaced by hyphens, and partially slugs
+  // albeit only in cases where the slug happened to match one of those.
+  // For backwards compatibility purposes we now support all three.
   Object.keys(originalSimpleIcons).forEach(key => {
-    const k = key.toLowerCase().replace(/ /g, '-')
-    simpleIcons[k] = originalSimpleIcons[key]
-    simpleIcons[k].base64 = {
-      default: svg2base64(
-        simpleIcons[k].svg.replace('<svg', `<svg fill="#${simpleIcons[k].hex}"`)
-      ),
-      light: svg2base64(
-        simpleIcons[k].svg.replace('<svg', `<svg fill="whitesmoke"`)
-      ),
-      dark: svg2base64(simpleIcons[k].svg.replace('<svg', `<svg fill="#333"`)),
+    const icon = originalSimpleIcons[key]
+    const title = icon.title.toLowerCase()
+    const legacyTitle = title.replace(/ /g, '-')
```

Comment:
should this be `title.toLowerCase().replace(/ /g, '-')` ?

Reply:
let me know if i've misunderstood, but `title` would already be lower cased as part of the initial assignment on the preceding line wouldn't it?

Reply:
There may be a compatibility problem here - the `title` may have changed in 5.0 as well. I tried before to use 5.0 icons and do the title transformation as what has been done here, but I actually had to do some extra processing to get it to work properly.

An example would be `Node.JS` (actually almost all badges with dot in title have this problem):

https://shields-staging-pr-6591.herokuapp.com/badge/foo-bar-blue?logo=node-dot-js
![](https://shields-staging-pr-6591.herokuapp.com/badge/foo-bar-blue?logo=node-dot-js)

https://shields.io/badge/foo-bar-blue?logo=node-dot-js
![](https://shields.io/badge/foo-bar-blue?logo=node-dot-js)


https://shields-staging-pr-6591.herokuapp.com/badge/foo-bar-blue?logo=node.js
![](https://shields-staging-pr-6591.herokuapp.com/badge/foo-bar-blue?logo=node.js)

https://shields.io/badge/foo-bar-blue?logo=node.js
![](https://shields.io/badge/foo-bar-blue?logo=node.js)

Reply:
@developStorm - you're referencing breaking changes that SimpleIcons made that's separate from the name vs. slug reference (more details in #6590). We don't attempt to maintain backwards compatibility when SI change their own slugs, and that's unrelated to this thread so going to hide these last two comments as off topic to keep the thread focused